File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lyricsscraper/GroupColumn.java
package com.wa2c.android.medoly.plugin.action.lyricsscraper;

/**
 * Group Column.
 */
public enum GroupColumn {
    /** ID. */
    _ID      ( "_ID"     , "_id"     , "INTEGER PRIMARY KEY AUTOINCREMENT" ),
    /** Group ID */
    GROUP_ID ("GROUP_ID" , "groupid" , "INTEGER NOT NULL" ),
    /** Group Name */
    NAME     ("NAME"     , "name"    , "TEXT NOT NULL" ),
    /** Group Name (ja) */
    NAME_JA  ("NAME_JA"  , "nameja"  , "TEXT NOT NULL" );

    /** Constructor. */
    GroupColumn(String name, String key, String constraint) {
        this.columnName = name;
        this.columnKey = key;
        this.constraint = constraint;
    }

    /** Column name. */
    private String columnName;
    /** Column key. */
    private String columnKey;
    /** Column constraint. */
    private String constraint;

    /** Get column name. */
    public String getColumnName() {
        return this.columnName;
    }
    /** Get column key. */
    public String getColumnKey() {
        return this.columnKey;
    }
    /** Get collumn constraint. */
    public String getConstraint() {
        return columnKey + " " + constraint;
    }
}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lyricsscraper/db/GroupColumn.kt
package com.wa2c.android.medoly.plugin.action.lyricsscraper.db

import java.util.Locale

/**
 * Group Column.
 */
enum class GroupColumn constructor(val columnName: String, val columnKey: String, private val constraint: String) {
    /** ID.  */
    _ID("_ID", "_id", "INTEGER PRIMARY KEY AUTOINCREMENT"),
    /** Group ID  */
    GROUP_ID("GROUP_ID", "groupid", "INTEGER NOT NULL"),
    /** Group Name  */
    NAME("NAME", "name", "TEXT NOT NULL"),
    /** Group Name (ja)  */
    NAME_JA("NAME_JA", "nameja", "TEXT NOT NULL");

    /** Get column constraint.  */
    fun getConstraint(): String {
        return "$columnKey $constraint"
    }

    companion object {

        fun findLocaleColumn(locale: Locale): GroupColumn {
            val colName = "NAME_" + locale.language.toUpperCase()
            for (c in GroupColumn.values()) {
                if (c.columnName == colName) {
                    return c
                }
            }
            return NAME
        }
    }

}
